cosmos/providers/databricks/notebook.py

```diff
@@ -91,8 +93,10 @@ def convert_to_databricks_workflow_task(
         self, relevant_upstreams: list[BaseOperator]
     ):
         """
-        Convert the operator to a Databricks workflow task that can be task in a workflow
+        Convert the operator to a Databricks workflow task that can be a task in a workflow
         """
+        if hasattr(self.task_group, "notebook_packages"):
+            self.notebook_packages.extend(self.task_group.notebook_packages)
```

Comment:
By using `extend` here, don't we have the risk of setting multiple times the same package with different versions at the workflow/job level?

From:
https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsCreate
```
The name of the PyPI package to install. An optional exact version specification is also supported. Examples: `simplejson and simplejson==3.8.0`. This field is required.
```
So we could have something like:
- Workflow level: `simplejson==3.8.0`
- Notebook level: `simplejson==2.0.0`

Would Databricks raise an exception? Or does it silently installs the last dependency definition (at the job level and not workflow)? It would be good to at least document this behaviour.

In future, depending on the usage, it may be worth raising a warning.

Reply:
yes, great point. Databricks does not allow duplicate installation of a library and fails the task as in the screenshot attached. Added a warning for the same in the docs.

<img width="1112" alt="Screenshot 2023-02-25 at 12 20 37 PM" src="https://user-images.githubusercontent.com/10206082/221343192-6ac9914e-b777-4dee-a43c-b0c2374195c9.png">